docs(stories): Add a note about using storyBook() with hooks and HoC wrappers (#1115)
getsentry/develop: src/docs/frontend/component-library.mdx

- Don't be afraid to import anything else you need to help illustrate use cases for your component. It's useful to leverage `<p>` and `<Fragment>` in most cases for example.
- If you are demonstrating a component that relies on a global hook like `useOrganization()` or `usePageFilters()`, then you should wrap your story with those context providers manually.
- Named exports, or a mix of named and default exports, are all supported. This is more useful with [non-framework stories](#non-framework-stories).

> - You can pass the component itself into `storyBook()` to get a nice title on the page. This avoids typos. But be careful of exceptions and use a string instead! For example, in a production build components that use `forwardRef` or HoC's can have minified names. Also hook names are always minified. Pass a string instead.

Reviewer: Appreciate the note! Might be nice to have in the code too since I know I would look there first and not realize there are docs here about it.

And just a little wording tweak:

```suggestion
- You can pass the component itself into `storyBook()` to get a nice title on the page. This avoids typos. But be careful of scenarios where a string should be used instead! For example, in a production build components that use `forwardRef` or HoC's can have minified names. Also hook names are always minified. Pass a string instead.
```
